test/org/rust/lang/inspections/RustInspectionsTest.kt

```diff
@@ -0,0 +1,18 @@
+package org.rust.lang.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import org.rust.lang.RustTestCase
+import kotlin.reflect.KClass
+
+class RustInspectionsTest : RustTestCase() {
+    override fun getTestDataPath() = "testData/org/rust/lang/inspections/fixtures"
+
+    private fun doTest(inspection: KClass<out LocalInspectionTool>) {
+        myFixture.enableInspections(inspection.java)
+        myFixture.testHighlighting(true, false, true, fileName)
+    }
+
+    fun testApproxConstant() = doTest(ApproxConstantInspection::class)
+    fun testSelfConvention() = doTest(SelfConventionInspection::class)
+    fun testPubSelfConvention() = doTest(PubSelfConventionInspection::class)
```

Comment:
Take a look at the [inline functions and reified type paremeters](https://kotlinlang.org/docs/reference/inline-functions.html#reified-type-parameters). This can be applicable here.